### ISSUE TITLE
Delete Unused Places Unexpectedly Performs Global Changes

### ISSUE THREAD

**Issue by miqrogroove** [fisharebest/webtrees#4448]
I was on the `map-data` page for my "New York" location and recently changed some tree data so that there were 2 places no longer in use.

When I clicked the Delete Unused Places button, webtrees purged 67 locations from my database.  😱

Some possible ways to improve this:

- Make changes context-sensitive.
- Make changes undoable.
- Add AYS

**Comment by fisharebest** [comment]
> When I clicked the Delete Unused Locations button, webtrees purged 67 locations from my database.

Perhaps these were all unused.  Perhaps your change ("New York" => "New York State"), caused lots of child places to become unused?

You didn't say "webtrees deleted a place that was in use", so I'm not sure what error you are reporting.

> Add AYS

AYS?

**Comment by miqrogroove** [comment]
> Perhaps these were all unused. Perhaps your change ("New York" => "New York State"), caused lots of child places to become unused?

Yes they were all unused.  No, it was not caused by the local change.  It performed a global purge of all unused locations in the database.  Simply unexpected behavior of the delete button on the New York page.


> > Add AYS
> 
> AYS?

"Are you sure" prompt is one way to avoid unexpected results.

**Comment by arbor95** [comment]
please close

**Comment by miqrogroove** [comment]
I see it's still labeled 'cannot reproduce'.  I'm not willing to test this again on my production site, so I would need to set up a 2nd instance for testing this bug.  Give me a couple more weeks to take care of that.

**Comment by fisharebest** [comment]
I shall close this now.  If you encounter something similar, please raise another issue.
